DIOSSession.m

```diff
@@ -67,13 +75,20 @@ - (NSMutableURLRequest *) requestWithMethod:(NSString *)method
                                  parameters:(NSDictionary *)parameters
 {
     NSString *urlString = [NSString stringWithFormat:@"%@/%@", [[self baseURL] absoluteString], path];
+    NSLog(@"Urlstring is %@", urlString);
     [self.requestSerializer setValue:@"application/json" forHTTPHeaderField:@"Accept"];
     [self.requestSerializer setValue:@"application/hal+json" forHTTPHeaderField:@"Content-Type"];
 
     if (self.signRequests) {
-        //doesntwork
+        
         [self.requestSerializer setAuthorizationHeaderFieldWithUsername:_basicAuthUsername password:_basicAuthPassword];
     }
+    else{
+        // If previously signRequest is set to YES then if temparoraly it has been set to NO then authorization header should be cleared
```

Comment:
I am explaining here ,
Consider the case where user is logged in to the Drupal via app i.e app is passing credentials on each request. But now if user has access to drupal through web and he/she changes the credentials there , in that case the app will receive 401 on request. So now app should no more supply credentials with request. I assume that boolean variable signRequest is there for this purpose. Or you can add one more helper method to remove the credentials from the shared session variable. I hope this clarifies.